src/Dictionaries/FlatDictionary.cpp

```diff
@@ -223,9 +238,11 @@ ColumnPtr FlatDictionary::getDescendants(
             parent_to_child[parent_key].emplace_back(static_cast<UInt64>(i));
     }
 
-    auto result = getKeysDescendantsArray(keys, parent_to_child, level);
+    size_t keys_found;
+    auto result = getKeysDescendantsArray(keys, parent_to_child, level, keys_found);
```

Comment:
If we fill parent_to_child map here, better to not move it into getKeysDescendantsArray function.

Reply:
But we don't filter this array with requested keys, am I missing something?

Reply:
Miscomment.